core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -78,4 +83,57 @@ public static CloseableIterable<CombinedScanTask> planTasks(CloseableIterable<Fi
             splitFiles),
         BaseCombinedScanTask::new);
   }
+
+  public static <T extends SplittableScanTask<T>> CloseableIterable<ScanTaskGroup<T>> planTaskGroups(
+      CloseableIterable<T> tasks,
+      long splitSize, int lookback, long openFileCost) {
+
+    Preconditions.checkArgument(splitSize > 0, "Invalid split size (negative or 0): %s", splitSize);
+    Preconditions.checkArgument(lookback > 0, "Invalid split planning lookback (negative or 0): %s", lookback);
+    Preconditions.checkArgument(openFileCost >= 0, "Invalid file open cost (negative): %s", openFileCost);
+
+    // capture manifests which can be closed after scan planning
+    CloseableIterable<T> splitTasks = CloseableIterable.combine(
+        FluentIterable.from(tasks).transformAndConcat(input -> input.split(splitSize)),
+        tasks);
+
+    Function<T, Long> weightFunc = task -> Math.max(task.totalSizeBytes(), task.totalFilesCount() * openFileCost);
```

Comment:
Should `totalSizeBytes` and `totalFilesCount` be part of `ScanTask` instead?

Reply:
Done.